### PR TITLE
Take default GC choices always

### PR DESCRIPTION
**Pull request by gbaraldi**
Revert only the automatic tuning of the GC because there are some issues with it. @d-netto is looking into them but it's probably not going to be done for 1.10

### CODE REVIEW

**Comment by gbaraldi** [comment]
@d-netto can you check if this doesn't make the performance too much worse?

File: src/gc.c [hunk 1]
```diff
@@ -3421,9 +3427,17 @@ static int _jl_gc_collect(jl_ptls_t ptls, jl_gc_collection_t collection)
         thrashing = 0; // maybe we should report this to the user or error out?
 
     int bad_result = (target_allocs*min_interval + heap_size) > 2 * jl_atomic_load_relaxed(&gc_heap_stats.heap_target); // Don't follow through on a bad decision
-    if (target_allocs == 0.0 || thrashing || bad_result) // If we are thrashing go back to default
-        target_allocs = 2*sqrt((double)heap_size/min_interval);
-    uint64_t target_heap = (uint64_t)target_allocs*min_interval + heap_size;
+    if (target_allocs == 0.0 || thrashing || bad_result || /*Always do the default to avoid issues with the algorithm*/ 1) // If we are thrashing go back to default
+        target_allocs =  2*sqrt((double)heap_size/min_interval);
```
Comment (NHDaly) [inline src/gc.c:3431]: In the MEMBALANCER case, this looks wrong: you're now missing the `*min_interval` multiplication on line 3440, right?
Should you have `target_heap *= min_interval` here?

Comment (gbaraldi) [reply src/gc.c:3431]: Right, thanks!

**Comment by d-netto** [comment]
We [backported](https://github.com/RelationalAI/julia/pull/89) this PR and ran our internal CI with it.

There seems to be OOM kills in a couple of tests.

**Comment by d-netto** [comment]
FWIW I can reproduce the issue reported on https://github.com/JuliaLang/julia/issues/51601 with this PR.

Peak RSS on 1.9 is around 400MB-500MB, and with this PR it goes beyond 60GB and fails on the serialization step of GCBenchmarks.

**Comment by vtjnash** [comment]
I don't know if you wanted to pull out the `jl_batch_accum_heap_size` code reorg but sounds like this wasn't sufficient and so didn't turn out to be useful

**Pull request closed without merging**